### ISSUE TITLE
Unexpected weight initialization

### ISSUE THREAD

**Issue by RangiLyu** [open-mmlab/mmengine#691]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmengine.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**

When using `Pretrained` init_cfg to init a model, the outside init_cfg will **disable all submodules' init_weights**.

If the outside model's pretrained weight does not strictly match, e.g. different num_classes, the mismatched submodule's weight will initialize **by pytorch's default init weight logic**, instead of the submodule's init_cfg. 

This lead's to some unacceptable results, e.g. a huge classification loss due to the failed initialization of the classification head when finetuning a model.

**Reproduction**

In mmdet, finetuning rtmdet_l_8xb32-300e_coco

```python
_base_ = './rtmdet_l_8xb32-300e_coco.py'

checkpoint = 'work_dir/rtmdet/rtmdet_l_8xb32-300e_coco/epoch_300.pth'
model = dict(bbox_head=dict(num_classes=1), init_cfg=dict(type='Pretrained', checkpoint=checkpoint))
```

**Comment by HAOCHENYE** [comment]
How about adding an `overwriten` argument to `PretrainedInit`? If `overwriten=False`(default value), `_is_init` of all submodules will be set to `True` during loading the pre-trained model, otherwise submodules could overwrite the loaded weight.

**Comment by Tuerte** [comment]
> Thanks for your error report and we appreciate it a lot.
> 
> **Checklist**
> 
> 1. I have searched related issues but cannot get the expected help.
> 2. I have read the [FAQ documentation](https://mmengine.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
> 3. The bug has not been fixed in the latest version.
> 
> **Describe the bug**
> 
> When using `Pretrained` init_cfg to init a model, the outside init_cfg will **disable all submodules' init_weights**.
> 
> If the outside model's pretrained weight does not strictly match, e.g. different num_classes, the mismatched submodule's weight will initialize **by pytorch's default init weight logic**, instead of the submodule's init_cfg.
> 
> This lead's to some unacceptable results, e.g. a huge classification loss due to the failed initialization of the classification head when finetuning a model.
> 
> **Reproduction**
> 
> In mmdet, finetuning rtmdet_l_8xb32-300e_coco
> 
> ```python
> _base_ = './rtmdet_l_8xb32-300e_coco.py'
> 
> checkpoint = 'work_dir/rtmdet/rtmdet_l_8xb32-300e_coco/epoch_300.pth'
> model = dict(bbox_head=dict(num_classes=1), init_cfg=dict(type='Pretrained', checkpoint=checkpoint))
> ``
hello, i use “”model = dict(bbox_head=dict(num_classes=1), init_cfg=dict(type='Pretrained', checkpoint=checkpoint))“”，but occurred some issues：
The model and loaded state dict do not match exactly
unexpected key in source state_dict: xxxxxxx（many unmatched layers）
missing keys in source state_dict:  xxxxxxx（many unmatched layers）
how to solute this? thansks.
